cibuildwheel/windows.py

```diff
@@ -102,7 +115,17 @@ def shell(args, env=None, cwd=None):
     for config in python_configurations:
         # install Python
         config_python_path = get_python_path(config)
-        simple_shell([nuget, "install"] + get_nuget_args(config))
+        if config.identifier.startswith('cp'):
+            simple_shell([nuget, "install"] + get_nuget_args(config))
+        elif config.identifier.startswith('pp') and not os.path.exists(config_python_path):
+            pypy_zip = os.path.join("C:\\cibw", config.url.rsplit('/', 1)[-1])
+            download(config.url, pypy_zip)
+            # Extract to the parent of config_python_path because the zip file still contains a directory
+            extract_zip(pypy_zip, os.path.dirname(config_python_path))
+            pypy_exe = 'pypy.exe' if config.version[0] == '2' else 'pypy3.exe'
+            simple_shell(['mklink', os.path.join(config_python_path, 'python.exe'), os.path.join(config_python_path, pypy_exe)])
```

Comment:
@joerick Are you OK making the symlinks in the directory into which we extracted PyPy? Because for macOS and the regular Python, you didn't want to create these symlinks in the installation directory?

Reply:
Yes, I'd slightly prefer we leave pypy as it was downloaded and set up our environment as we want it externally, using a separate PATH entry... on macOS that's already set up as `/tmp/cibw_bin`, but I guess we'd have to add something similar here.

Reply:
@joerick I've done so for macOS, and cleaned up a bit more; do have a look at the last commit, please.

However, on Windows, it doesn't seem that easy (as demonstrated by that line "YannickJadoul force-pushed the YannickJadoul:pypy branch 11 times, most recently from bc7665c to ef329c0 2 hours ago")

I still don't fully understand what goes wrong, but somehow, it seems that when Python is invoked through a Windows symlink, something goes wrong with finding the right `pip`/installed libraries. Do have a try yourself if you want to have a go. Maybe I've overlooked something silly. But after 2 hours of staring at Azure Pipeline logs and writing debug output and trying new things, I'm personally just as happy leaving these two symlinks inside of PyPy's installation dir :-D

Reply:
Fair enough! :)